// @joshfarrant/shortcuts-js/build/utils/buildShortcutTemplate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const flatten_1 = require("./flatten");
/** @ignore */
exports.buildShortcutTemplate = (actions = [], options) => ({
    WFWorkflowClientVersion: '724',
    WFWorkflowClientRelease: '2.1',
    WFWorkflowIcon: {
        WFWorkflowIconStartColor: options.icon.color,
        WFWorkflowIconGlyphNumber: options.icon.glyph,
    },
    WFWorkflowImportQuestions: [],
    WFWorkflowTypes: [
        'WatchKit',
        ...(options.showInWidget ? ['NCWidget'] : []),
    ],
    WFWorkflowInputContentItemClasses: [
        'WFAppStoreAppContentItem',
        'WFArticleContentItem',
        'WFContactContentItem',
        'WFDateContentItem',
        'WFEmailAddressContentItem',
        'WFGenericFileContentItem',
        'WFImageContentItem',
        'WFiTunesProductContentItem',
        'WFLocationContentItem',
        'WFDCMapsLinkContentItem',
        'WFAVAssetContentItem',
        'WFPDFContentItem',
        'WFPhoneNumberContentItem',
        'WFRichTextContentItem',
        'WFSafariWebPageContentItem',
        'WFStringContentItem',
        'WFURLContentItem',
    ],
    WFWorkflowActions: flatten_1.flatten(actions),
});
