// @joshfarrant/shortcuts-js/build/actions/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const addToReadingList_1 = require("./addToReadingList");
exports.addToReadingList = addToReadingList_1.default;
const addToVariable_1 = require("./addToVariable");
exports.addToVariable = addToVariable_1.default;
const airDrop_1 = require("./airDrop");
exports.airDrop = airDrop_1.default;
const ask_1 = require("./ask");
exports.ask = ask_1.default;
const base64Encode_1 = require("./base64Encode");
exports.base64Encode = base64Encode_1.default;
const calculate_1 = require("./calculate");
exports.calculate = calculate_1.default;
const calculateStatistics_1 = require("./calculateStatistics");
exports.calculateStatistics = calculateStatistics_1.default;
const call_1 = require("./call");
exports.call = call_1.default;
const changeCase_1 = require("./changeCase");
exports.changeCase = changeCase_1.default;
const chooseFromMenu_1 = require("./chooseFromMenu");
exports.chooseFromMenu = chooseFromMenu_1.default;
const clearUpNext_1 = require("./clearUpNext");
exports.clearUpNext = clearUpNext_1.default;
const comment_1 = require("./comment");
exports.comment = comment_1.default;
const conditional_1 = require("./conditional");
exports.conditional = conditional_1.default;
const continueShortcutInApp_1 = require("./continueShortcutInApp");
exports.continueShortcutInApp = continueShortcutInApp_1.default;
const correctSpelling_1 = require("./correctSpelling");
exports.correctSpelling = correctSpelling_1.default;
const count_1 = require("./count");
exports.count = count_1.default;
const createAlarm_1 = require("./createAlarm");
exports.createAlarm = createAlarm_1.default;
const createNote_1 = require("./createNote");
exports.createNote = createNote_1.default;
const date_1 = require("./date");
exports.date = date_1.default;
const deletePhotos_1 = require("./deletePhotos");
exports.deletePhotos = deletePhotos_1.default;
const detectLanguageWithMicrosoft_1 = require("./detectLanguageWithMicrosoft");
exports.detectLanguageWithMicrosoft = detectLanguageWithMicrosoft_1.default;
const dictionary_1 = require("./dictionary");
exports.dictionary = dictionary_1.default;
const exitShortcut_1 = require("./exitShortcut");
exports.exitShortcut = exitShortcut_1.default;
const expandURL_1 = require("./expandURL");
exports.expandURL = expandURL_1.default;
const extractArchive_1 = require("./extractArchive");
exports.extractArchive = extractArchive_1.default;
const facetime_1 = require("./facetime");
exports.facetime = facetime_1.default;
const formatDate_1 = require("./formatDate");
exports.formatDate = formatDate_1.default;
const generateHash_1 = require("./generateHash");
exports.generateHash = generateHash_1.default;
const getAddressesFromInput_1 = require("./getAddressesFromInput");
exports.getAddressesFromInput = getAddressesFromInput_1.default;
const getBatteryLevel_1 = require("./getBatteryLevel");
exports.getBatteryLevel = getBatteryLevel_1.default;
const getClipboard_1 = require("./getClipboard");
exports.getClipboard = getClipboard_1.default;
const getContactsFromInput_1 = require("./getContactsFromInput");
exports.getContactsFromInput = getContactsFromInput_1.default;
const getContentsOfURL_1 = require("./getContentsOfURL");
exports.getContentsOfURL = getContentsOfURL_1.default;
const getContentsOfWebPage_1 = require("./getContentsOfWebPage");
exports.getContentsOfWebPage = getContentsOfWebPage_1.default;
const getCurrentIPAddress_1 = require("./getCurrentIPAddress");
exports.getCurrentIPAddress = getCurrentIPAddress_1.default;
const getCurrentLocation_1 = require("./getCurrentLocation");
exports.getCurrentLocation = getCurrentLocation_1.default;
const getCurrentSong_1 = require("./getCurrentSong");
exports.getCurrentSong = getCurrentSong_1.default;
const getDatesFromInput_1 = require("./getDatesFromInput");
exports.getDatesFromInput = getDatesFromInput_1.default;
const getDeviceDetails_1 = require("./getDeviceDetails");
exports.getDeviceDetails = getDeviceDetails_1.default;
const getDictionaryFromInput_1 = require("./getDictionaryFromInput");
exports.getDictionaryFromInput = getDictionaryFromInput_1.default;
const getDictionaryValue_1 = require("./getDictionaryValue");
exports.getDictionaryValue = getDictionaryValue_1.default;
const getDiffbotArticleFromWebPage_1 = require("./getDiffbotArticleFromWebPage");
exports.getDiffbotArticleFromWebPage = getDiffbotArticleFromWebPage_1.default;
const getEmailAddressesFromInput_1 = require("./getEmailAddressesFromInput");
exports.getEmailAddressesFromInput = getEmailAddressesFromInput_1.default;
const getFramesFromImage_1 = require("./getFramesFromImage");
exports.getFramesFromImage = getFramesFromImage_1.default;
const getHeadersOfURL_1 = require("./getHeadersOfURL");
exports.getHeadersOfURL = getHeadersOfURL_1.default;
const getImagesFromInput_1 = require("./getImagesFromInput");
exports.getImagesFromInput = getImagesFromInput_1.default;
const getLastImport_1 = require("./getLastImport");
exports.getLastImport = getLastImport_1.default;
const getLatestBursts_1 = require("./getLatestBursts");
exports.getLatestBursts = getLatestBursts_1.default;
const getLatestLivePhotos_1 = require("./getLatestLivePhotos");
exports.getLatestLivePhotos = getLatestLivePhotos_1.default;
const getLatestScreenshots_1 = require("./getLatestScreenshots");
exports.getLatestScreenshots = getLatestScreenshots_1.default;
const getLatestVideos_1 = require("./getLatestVideos");
exports.getLatestVideos = getLatestVideos_1.default;
const getLinkToFile_1 = require("./getLinkToFile");
exports.getLinkToFile = getLinkToFile_1.default;
const getMapsURL_1 = require("./getMapsURL");
exports.getMapsURL = getMapsURL_1.default;
const getMyShortcuts_1 = require("./getMyShortcuts");
exports.getMyShortcuts = getMyShortcuts_1.default;
const getName_1 = require("./getName");
exports.getName = getName_1.default;
const getNameOfEmoji_1 = require("./getNameOfEmoji");
exports.getNameOfEmoji = getNameOfEmoji_1.default;
const getNetworkDetails_1 = require("./getNetworkDetails");
exports.getNetworkDetails = getNetworkDetails_1.default;
const getPhoneNumbersFromInput_1 = require("./getPhoneNumbersFromInput");
exports.getPhoneNumbersFromInput = getPhoneNumbersFromInput_1.default;
const getTextFromInput_1 = require("./getTextFromInput");
exports.getTextFromInput = getTextFromInput_1.default;
const getTimeBetweenDates_1 = require("./getTimeBetweenDates");
exports.getTimeBetweenDates = getTimeBetweenDates_1.default;
const getType_1 = require("./getType");
exports.getType = getType_1.default;
const getURLsFromInput_1 = require("./getURLsFromInput");
exports.getURLsFromInput = getURLsFromInput_1.default;
const getVariable_1 = require("./getVariable");
exports.getVariable = getVariable_1.default;
const list_1 = require("./list");
exports.list = list_1.default;
const makeArchive_1 = require("./makeArchive");
exports.makeArchive = makeArchive_1.default;
const makeMarkdownFromRichText_1 = require("./makeMarkdownFromRichText");
exports.makeMarkdownFromRichText = makeMarkdownFromRichText_1.default;
const makeRichTextFromHTML_1 = require("./makeRichTextFromHTML");
exports.makeRichTextFromHTML = makeRichTextFromHTML_1.default;
const makeRichTextFromMarkdown_1 = require("./makeRichTextFromMarkdown");
exports.makeRichTextFromMarkdown = makeRichTextFromMarkdown_1.default;
const markup_1 = require("./markup");
exports.markup = markup_1.default;
const matchText_1 = require("./matchText");
exports.matchText = matchText_1.default;
const nothing_1 = require("./nothing");
exports.nothing = nothing_1.default;
const number_1 = require("./number");
exports.number = number_1.default;
const openApp_1 = require("./openApp");
exports.openApp = openApp_1.default;
const openInBooks_1 = require("./openInBooks");
exports.openInBooks = openInBooks_1.default;
const openURLs_1 = require("./openURLs");
exports.openURLs = openURLs_1.default;
const openURLsInChrome_1 = require("./openURLsInChrome");
exports.openURLsInChrome = openURLsInChrome_1.default;
const pauseMusic_1 = require("./pauseMusic");
exports.pauseMusic = pauseMusic_1.default;
const playSound_1 = require("./playSound");
exports.playSound = playSound_1.default;
const postOnFacebook_1 = require("./postOnFacebook");
exports.postOnFacebook = postOnFacebook_1.default;
const postOnInstagram_1 = require("./postOnInstagram");
exports.postOnInstagram = postOnInstagram_1.default;
const postToTumblr_1 = require("./postToTumblr");
exports.postToTumblr = postToTumblr_1.default;
const postToWordPress_1 = require("./postToWordPress");
exports.postToWordPress = postToWordPress_1.default;
const print_1 = require("./print");
exports.print = print_1.default;
const quickLook_1 = require("./quickLook");
exports.quickLook = quickLook_1.default;
const randomNumber_1 = require("./randomNumber");
exports.randomNumber = randomNumber_1.default;
const removeReminders_1 = require("./removeReminders");
exports.removeReminders = removeReminders_1.default;
const repeat_1 = require("./repeat");
exports.repeat = repeat_1.default;
const runHomeScene_1 = require("./runHomeScene");
exports.runHomeScene = runHomeScene_1.default;
const runJavaScriptOnWebPage_1 = require("./runJavaScriptOnWebPage");
exports.runJavaScriptOnWebPage = runJavaScriptOnWebPage_1.default;
const runScriptOverSSH_1 = require("./runScriptOverSSH");
exports.runScriptOverSSH = runScriptOverSSH_1.default;
const runShortcut_1 = require("./runShortcut");
exports.runShortcut = runShortcut_1.default;
const scanQROrBarcode_1 = require("./scanQROrBarcode");
exports.scanQROrBarcode = scanQROrBarcode_1.default;
const selectEmailAddress_1 = require("./selectEmailAddress");
exports.selectEmailAddress = selectEmailAddress_1.default;
const selectPhoneNumber_1 = require("./selectPhoneNumber");
exports.selectPhoneNumber = selectPhoneNumber_1.default;
const setAirplaneMode_1 = require("./setAirplaneMode");
exports.setAirplaneMode = setAirplaneMode_1.default;
const setBluetooth_1 = require("./setBluetooth");
exports.setBluetooth = setBluetooth_1.default;
const setBrightness_1 = require("./setBrightness");
exports.setBrightness = setBrightness_1.default;
const setCellularData_1 = require("./setCellularData");
exports.setCellularData = setCellularData_1.default;
const setDictionaryValue_1 = require("./setDictionaryValue");
exports.setDictionaryValue = setDictionaryValue_1.default;
const setDoNotDisturb_1 = require("./setDoNotDisturb");
exports.setDoNotDisturb = setDoNotDisturb_1.default;
const setLowPowerMode_1 = require("./setLowPowerMode");
exports.setLowPowerMode = setLowPowerMode_1.default;
const setName_1 = require("./setName");
exports.setName = setName_1.default;
const setTorch_1 = require("./setTorch");
exports.setTorch = setTorch_1.default;
const setVariable_1 = require("./setVariable");
exports.setVariable = setVariable_1.default;
const setVolume_1 = require("./setVolume");
exports.setVolume = setVolume_1.default;
const setWiFi_1 = require("./setWiFi");
exports.setWiFi = setWiFi_1.default;
const share_1 = require("./share");
exports.share = share_1.default;
const shareWithExtensions_1 = require("./shareWithExtensions");
exports.shareWithExtensions = shareWithExtensions_1.default;
const showAlert_1 = require("./showAlert");
exports.showAlert = showAlert_1.default;
const showDefinition_1 = require("./showDefinition");
exports.showDefinition = showDefinition_1.default;
const showInCalendar_1 = require("./showInCalendar");
exports.showInCalendar = showInCalendar_1.default;
const showInITunesStore_1 = require("./showInITunesStore");
exports.showInITunesStore = showInITunesStore_1.default;
const showInMaps_1 = require("./showInMaps");
exports.showInMaps = showInMaps_1.default;
const showNotification_1 = require("./showNotification");
exports.showNotification = showNotification_1.default;
const showResult_1 = require("./showResult");
exports.showResult = showResult_1.default;
const skipBack_1 = require("./skipBack");
exports.skipBack = skipBack_1.default;
const skipForward_1 = require("./skipForward");
exports.skipForward = skipForward_1.default;
const speakText_1 = require("./speakText");
exports.speakText = speakText_1.default;
const text_1 = require("./text");
exports.text = text_1.default;
const trimMedia_1 = require("./trimMedia");
exports.trimMedia = trimMedia_1.default;
const tweet_1 = require("./tweet");
exports.tweet = tweet_1.default;
const URL_1 = require("./URL");
exports.URL = URL_1.default;
const URLEncode_1 = require("./URLEncode");
exports.URLEncode = URLEncode_1.default;
const vibrateDevice_1 = require("./vibrateDevice");
exports.vibrateDevice = vibrateDevice_1.default;
const viewContentGraph_1 = require("./viewContentGraph");
exports.viewContentGraph = viewContentGraph_1.default;
const wait_1 = require("./wait");
exports.wait = wait_1.default;
const waitToReturn_1 = require("./waitToReturn");
exports.waitToReturn = waitToReturn_1.default;
// 3rd-party app imports
const pythonista = require("./pythonista");
exports.pythonista = pythonista;
